homeassistant/components/snmp/sensor.py

```diff
@@ -106,6 +108,13 @@ async def async_setup_platform(
     default_value = config.get(CONF_DEFAULT_VALUE)
     unique_id = config.get(CONF_UNIQUE_ID)
 
+    try:
+        # Try IPv4 first.
+        target = UdpTransportTarget((host, port), timeout=DEFAULT_TIMEOUT)
+    except PySnmpError as e:
+        # Then try IPv6.
+        target = Udp6TransportTarget((host, port), timeout=DEFAULT_TIMEOUT)
```

Comment:
This can also raise `PySnmpError`.

Reply:
Indeed, though this was previously unhandled so I figured it wouldn't hurt to keep that part as it was.
Anyway, I've now updated the commit to catch it too and instead log an error.